### PR TITLE
chore: use let and const

### PR DESCRIPTION
**Pull request by JustinBeckwith**

### CODE REVIEW

File: src/connection-pool.js [hunk 1]
```diff
@@ -467,21 +467,20 @@ ConnectionPool.prototype.pause = function() {
...
   }
-
-  var timeoutHandle = setTimeout(createConnection, delay);
+  const timeoutHandle = setTimeout(createConnection, delay);
   this.queue.push(timeoutHandle);
-
   function createConnection() {
-    self.createConnection();
-    self.queue.splice(self.queue.indexOf(timeoutHandle), 1);
+    setImmediate(() => {
```
[the remark this answers is not in the document]
Comment (kinwa91) [reply src/connection-pool.js]: Variable hoisting:
https://medium.freecodecamp.org/what-is-variable-hoisting-differentiating-between-var-let-and-const-in-es6-f1a70bb43d

[the remark this answers is not in the document]
Comment (crwilcox) [reply src/connection-pool.js]: I think the issue may be that timeoutHandle needs the function createConnection and createConnection uses timeoutHandle. I think when attempts are made to order instantiation of these it is getting caught up. It doesn't matter which it puts first, the other is going to be missing a reference at that time.

I believe by wrapping the inner in setImmediate the eval is delayed and there is no ordering issue anymore.

So, TL;DR - I sort of think your solution is fine. Though I don't like it.

**Review by crwilcox — APPROVED** [review]
